src/main/java/com/jamieswhiteshirt/clothesline/mixin/world/WorldMixin.java

```diff
@@ -1,38 +1,38 @@
-        if (CollisionView.super.canPlace(state, pos, shapeContext)) {
-            VoxelShape shape = state.getCollisionShape(this, pos);
-            if (!shape.isEmpty()) {
-                net.minecraft.util.math.Box bb = shape.offset(pos.getX(), pos.getY(), pos.getZ()).getBoundingBox();
-                com.jamieswhiteshirt.rtree3i.Box box = com.jamieswhiteshirt.rtree3i.Box.create(
-                    MathHelper.floor(bb.minX), MathHelper.floor(bb.minY), MathHelper.floor(bb.minZ),
-                    MathHelper.ceil(bb.maxX), MathHelper.ceil(bb.maxY), MathHelper.ceil(bb.maxZ)
-                );
-                boolean intersects = getNetworkManager().getNetworks().getEdges()
-                    .values(box::intersectsClosed)
-                    .anyMatch(edge -> {
-                        Line line = edge.getPathEdge().getLine();
-                        return shape.rayTrace(line.getFromVec(), line.getToVec(), pos) != null;
-                    });
-                return !intersects;
-            }
-            return true;
-        }
-        return false;
-    }
-}
+package com.jamieswhiteshirt.clothesline.mixin.world;
+
+import com.jamieswhiteshirt.clothesline.api.Line;
+import com.jamieswhiteshirt.clothesline.api.NetworkManagerProvider;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.ShapeContext;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.shape.VoxelShape;
+import net.minecraft.world.CollisionView;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(World.class)
+public abstract class WorldMixin implements CollisionView, NetworkManagerProvider {
+    @Override
+    public boolean canPlace(BlockState state, BlockPos pos, ShapeContext shapeContext) {
+        if (CollisionView.super.canPlace(state, pos, shapeContext)) {
+            VoxelShape shape = state.getCollisionShape(this, pos);
+            if (!shape.isEmpty()) {
+                net.minecraft.util.math.Box bb = shape.offset(pos.getX(), pos.getY(), pos.getZ()).getBoundingBox();
+                com.jamieswhiteshirt.rtree3i.Box box = com.jamieswhiteshirt.rtree3i.Box.create(
+                    MathHelper.floor(bb.minX), MathHelper.floor(bb.minY), MathHelper.floor(bb.minZ),
+                    MathHelper.ceil(bb.maxX), MathHelper.ceil(bb.maxY), MathHelper.ceil(bb.maxZ)
+                );
+                boolean intersects = getNetworkManager().getNetworks().getEdges()
+                    .values(box::intersectsClosed)
+                    .anyMatch(edge -> {
+                        Line line = edge.getPathEdge().getLine();
+                        return shape.raycast(line.getFromVec(), line.getToVec(), pos) != null;
+                    });
+                return !intersects;
+            }
+            return true;
+        }
+        return false;
+    }
+}

```

Comment:
When you do this "Add files via upload", it made a patch that completely overwrites the file instead of changing the one line that is affected.

Reply:
i apologize for my incorrectly changing it, only line 30 was changed since mappings moved to use "raycast" instead of "rayTrace"